client/layout/navigation.js

```diff
@@ -22,15 +24,24 @@ import { isWCAdmin } from '../dashboard/utils';
 import { addHistoryListener } from '../navigation/utils';
 
 const NavigationPlugin = () => {
-	const [ persistedQuery, setPersistedQuery ] = useState(
-		getPersistedQuery()
-	);
+	const { persistedQuery } = useSelect( ( select ) => {
+		return {
+			persistedQuery: select(
+				NAVIGATION_STORE_NAME
+			).getPersistedQueryFromStore(),
+		};
+	} );
+
+	const { setPersistedQuery } = useDispatch( NAVIGATION_STORE_NAME );
 
 	const pathIsExcluded = ( path ) =>
 		getQueryExcludedScreens().includes( getScreenFromPath( path ) );
 
 	// Update the persisted queries when history is updated
 	useEffect( () => {
+		if ( Object.keys( getPersistedQuery() ).length ) {
+			setPersistedQuery( getPersistedQuery() );
```

Comment:
Event driven setting resolves this, but if we do plan to use this in other components we would want to make sure this is not only set in the navigation as it could be toggled off.